trio/_channel.py

```diff
@@ -109,28 +137,28 @@ def statistics(self):
 
 
 @attr.s(eq=False, repr=False)
-class MemorySendChannel(SendChannel, metaclass=NoPublicConstructor):
-    _state = attr.ib()
-    _closed = attr.ib(default=False)
+class MemorySendChannel(SendChannel, Generic[SendType], metaclass=NoPublicConstructor):
```

Comment:
Should `SendChannel` be generic?

Reply:
the abc has generic, so I think you're right that the type should be passed through to it as well here.

Reply:
Ye, if the type doesn't get passed through then this code raises no errors on the assignment to bar:
```python
foo: MemorySendChannel[int] = ...
bar: SendChannel[str] = foo
```